### PR TITLE
Update side panel tab dynamically

### PR DESCRIPTION
**Pull request by ajbozarth**
Currently the side panel tab is only set when opening the panel.

This changes it to update the tab whenever the selected nodes change,
following the same rules as it previously used when opening the panel.

Fixes https://github.com/elyra-ai/elyra/issues/1827

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/PipelineEditor/index.tsx [hunk 1]
```diff
@@ -428,13 +428,22 @@ const PipelineEditor = forwardRef(
...
+      }
+    };
+
     const handleClickAction = useCallback(
       (e: CanvasClickEvent) => {
         if (e.clickType === "DOUBLE_CLICK" && e.objectType === "node") {
           if (onDoubleClickNode !== undefined) {
             return onDoubleClickNode(e);
           }
-          setCurrentTab("properties");
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineEditor/index.tsx:437]: why is this removed?

File: packages/pipeline-editor/src/PipelineEditor/index.tsx [hunk 2]
```diff
@@ -444,6 +453,7 @@ const PipelineEditor = forwardRef(
     const [selectedNodeIDs, setSelectedNodeIDs] = useState<string[]>();
     const handleSelectionChange = useCallback((e: CanvasSelectionEvent) => {
       setSelectedNodeIDs(e.selectedNodes.map((n: NodeTypeDef) => n.id));
+      setPropertiesTab(e.selectedNodes.length > 0);
     }, []);
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineEditor/index.tsx:457]: you either need to either add `setPropertiesTab` as a dependency or move the logic or function here. I would just move the logic here since nothing else uses this function

**Review by bourdakos1 — CHANGES_REQUESTED** [review]

File: packages/pipeline-editor/src/PipelineEditor/index.tsx (hunk 2 above)
Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineEditor/index.tsx:457]: I'll move the logic, I had it in a function because I was calling it in a bunch of places, before I realized that calling it here covered every case.

File: packages/pipeline-editor/src/PipelineEditor/index.tsx (hunk 1 above)
Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineEditor/index.tsx:437]: I noticed it was redundant while I was working, the first line of the function call below does this.

Comment (bourdakos1) [reply packages/pipeline-editor/src/PipelineEditor/index.tsx:437]: this opens the properties panel when you double click on a node, you wouldn't notice it in elyra because we override the functionality with `onDoubleClickNode` to open the file instead

Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineEditor/index.tsx:437]: I figured that's what it is for, what I meant was the first line in the properties case in the editActionHandler call in the following line is literally the same line as this. So when the code hits this it will always call setCurrentTab twice. So I removed the redundant line here

Comment (bourdakos1) [reply packages/pipeline-editor/src/PipelineEditor/index.tsx:437]: Oooohhh I see :) I didn’t see that as part of the diff, I thought you were referring to the new logic in selection change handler

nice catch

**Review by bourdakos1 — APPROVED** [review]
LGTM
